torch/_subclasses/fake_tensor.py

```diff
@@ -922,6 +920,12 @@ def tolist(self) -> Any:
         else:
             return [elem.tolist() for elem in self]
 
+    def detach(self) -> torch.Tensor:  # type: ignore[override]
```

Comment:
overriding detach() for FakeTensor generally just to support NJT feels a bit sus. For example, this code will not run if detach() is invoked from C++. Why do we need it?

Reply:
> Why do we need it?

We need a way to update the tensor registry for the newly created offsets instance to have the same tensor id as the original offsets. 

> overriding detach() for FakeTensor generally just to support NJT feels a bit sus. 

Agreed, since this is specific to view_avoid_dupes_with_primals, I guess it makes more sense to keep the extra local to there as well.